Src/Support/GoogleApis.Auth.DotNet4/OAuth2/AuthorizationDialog.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Google.Apis.Auth.OAuth2.Responses;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    public partial class AuthorizationDialog : Form
+    {
+        private AuthorizationCodeResponseUrl responseUrl;
+
+        public AuthorizationCodeResponseUrl ResponseUrl
+        {
+            get
+            {
+                return responseUrl;
+            }
+        }
+        /// <summary> Constructs a new authentication broker dialog. </summary>
+        public AuthorizationDialog(Uri authUri)
+        {
+            InitializeComponent();
+            // Set the browser to the initial uri.
+            authWebBrowser.Navigate(authUri);
+            // Need to track when the results has been received.
+            authWebBrowser.DocumentCompleted += authWebBrowser_DocumentCompleted;
```

Comment:
If I had used a WPF Window instead, would you have the same concern?
       Yes, adding any UI dependency to the DLL would have given me the same concern.   IMO the client lib should not be for UI it should just be for the background functionality of accessing the APIs. 

I must respectfully challenge the statement that Windows Forms is not something a lot of people are going to be using.
      I only use windows forms, my wording is probably off.   While I have used WPF 90% of what i do is either console apps or win forms.    

Unfortunately my .Net skills are not IMO up to the challenge of adding a change like this, nor can I comment on your approach.  I will leave that to the .net gurus on the client lib.    However I like the sounds of your BYOB idea.  

BTW:   I made my own class to handle Auth I can then feed the refresh token to the Datastore manually.   [Google APIS auth Simple](http://www.daimto.com/google-apis-auth-simple/)  basically because I needed to be able to launch my own browser from forms.